aiven_mysql_migrate/utils.py

```diff
@@ -49,13 +61,31 @@ def from_uri(uri: str, name: Optional[str] = None):
         port = res.port or DEFAULT_MYSQL_PORT
         options = parse_qs(res.query)
         ssl = not (options and options.get("ssl-mode", ["DISABLE"]) == ["DISABLE"])
-        return MySQLConnectionInfo(
-            hostname=res.hostname, port=port, username=res.username, password=res.password, ssl=ssl, name=name
-        )
+
+        LOGGER.debug("from_uri - sslca:[%s], sslcert:[%s], sslkey:[%s]", sslca, sslcert, sslkey)
+        if sslca and sslcert and sslkey:
+            return MySQLConnectionInfo(
+                hostname=res.hostname,
+                port=port,
+                username=res.username,
+                password=res.password,
+                ssl=ssl,
+                sslca=sslca,
+                sslcert=sslcert,
+                sslkey=sslkey,
+                name=name
+            )
+        else:
+            return MySQLConnectionInfo(
+                hostname=res.hostname, port=port, username=res.username, password=res.password, ssl=ssl, name=name
+            )
 
     def to_uri(self):
         ssl_mode = "DISABLE" if not self.ssl else "REQUIRE"
-        return f"mysql://{self.username}:{self.password}@{self.hostname}:{self.port}/?ssl-mode={ssl_mode}"
+        ssl_auth = f"&ssl-ca={self.sslca}&ssl-cert={self.sslcert}&ssl-key={self.sslkey}" \
```

Comment:
no url-encoding? Can you be sure all chars are urlsafe?

Reply:
Thanks for catching this! fixed it!